src/main/java/org/folio/services/invoice/InvoicePaymentService.java

```diff
@@ -49,10 +57,30 @@ public Future<Void> payInvoice(Invoice invoice, List<InvoiceLine> invoiceLines,
     invoice.setPaymentDate(invoice.getMetadata().getUpdatedDate());
     return getInvoiceWorkflowDataHolders(invoice, invoiceLines, requestContext)
       .compose(holders -> paymentCreditWorkflowService.handlePaymentsAndCreditsCreation(holders, requestContext))
-      .compose(vVoid -> CompositeFuture.join(payPoLines(invoiceLines, requestContext), voucherService.payInvoiceVoucher(invoice.getId(), requestContext)))
+      .compose(vVoid -> CompositeFuture.join(updatePoLinesStatus(invoice, invoiceLines, requestContext), voucherService.payInvoiceVoucher(invoice.getId(), requestContext)))
       .mapEmpty();
   }
 
+  private Future<Void> updatePoLinesStatus(Invoice invoice, List<InvoiceLine> invoiceLines, RequestContext requestContext) {
+    String fundID = invoiceLines.stream()
+      .flatMap(invoiceLine -> invoiceLine.getFundDistributions().stream())
+      .map(FundDistribution::getFundId)
+      .findFirst()
+      .orElse(null);
+
+    if (StringUtils.isNotBlank(fundID)) {
+      return currentFiscalYearService.getCurrentFiscalYearByFund(fundID, requestContext)
+        .compose(currentFiscalYear -> {
+          if (Objects.equals(currentFiscalYear.getId(), invoice.getFiscalYearId())) {
```

Comment:
I think `invoice.getFiscalYearId()` can be `null`, in which case the previous behavior should be applied (with no additional call).

Reply:
I think we should change this field to required after changing the UI.

Reply:
> I think we should change this field to required after changing the UI.

in that case we should also create migration script which will populate this field for existing invoices. And I am not sure if it is possible

Reply:
Right, the fiscal year information would be in mod-finance-storage as a transaction, assuming the invoice lines are linked to order lines (which is not even guaranteed), or if we just want to use the current fiscal year it would also be in a mod-finance-storage table. We would not be able to use that information from a migration script in mod-invoice. So we should handle the case of a null fiscalYearId. This will also make it simpler for the UI code.